qp_deblur/deblur.py

```diff
@@ -73,20 +74,69 @@ def generate_deblur_workflow_commands(preprocessed_fp, out_dir, parameters):
     return cmd
 
 
-def generate_sepp_placements(seqs):
+def generate_sepp_placements(seqs, out_dir, threads, reference_phylogeny=None,
+                             reference_alignment=None):
     """Generates the sepp commands
 
     Parameters
     ----------
     seqs : list of str
         A list of seqs to generate placements
+    out_dir : str
+        The job output directory
+    threads : int
+        Number if CPU cores to use
+    reference_phylogeny : str, optional
+        A filepath to an alternative reference phylogeny for SEPP.
+        If None, default phylogeny is uses, which is Greengenes 13.8 99% id.
+    reference_alignment : str, optional
+        A filepath to an alternative reference alignment for SEPP.
+        If None, default alignment is uses, which is Greengenes 13.8 99% id.
 
     Returns
     -------
     dict of strings
         keys are the seqs, values are the new placements as JSON strings
     """
-    return {}
+    # return an empty dict if no sequences have been passed to the function
+    if len(seqs) < 1:
+        return {}
+
+    # Create a multiple fasta file for all input seqs
+    file_input = "%s/input.fasta" % out_dir
+    with open(file_input, 'w') as fh_input:
+        for seq in seqs:
+            fh_input.write(">%s\n%s\n" % (seq, seq))
+
+    # execute SEPP
+    run_name = 'qiita'
+    param_phylogeny = ''
+    if reference_phylogeny is not None:
+        param_phylogeny = ' -t %s ' % abspath(reference_phylogeny)
+    param_alignment = ''
+    if reference_alignment is not None:
+        param_alignment = ' -a %s ' % abspath(reference_alignment)
+    # SEPP writes output into the current working directory (cwd), therefore
+    # we here first need to store the cwd, then move into the output directory,
+    # perform SEPP and move back to the stored cwd for a clean state
+    curr_pwd = environ['PWD']
+    std_out, std_err, return_value = system_call(
+        'cd %s && run-sepp.sh %s %s -x %i %s %s; cd %s' %
+        (out_dir, file_input, run_name, threads,
+         param_phylogeny, param_alignment, curr_pwd))
+
+    # parse placements from SEPP results
+    try:
+        with open('%s/%s_placement.json' % (out_dir, run_name),
+                  'r') as fh_placements:
+            plcmnts = json.loads(fh_placements.read())
+            return {p['nm'][0][0]: p['p'] for p in plcmnts['placements']}
```

Comment:
From the tests I'm not sure which is the format that this is returned. I'm wondering if the current format is not extensible as we were discussing earlier. Do you mind adding a test in which the format of this (at least for one key) is tested?

Reply:
sure thing. It's now in